### ISSUE TITLE
First mutation

### ISSUE THREAD

**Issue by oliviertassinari** [reactjs/react-router-redux#193]
I'm using the latest version of this repository.
The `@@router/UPDATE_LOCATION` action do not reach the end of my middleware chain during the first mutation of the state.
I'm wondering if this is intentional, or an issue with this project (it's working with `redux-router`).

Here is my middleware chain in case you ask:
```js
import promiseMiddleware from 'redux-promise';
import thunk from 'redux-thunk';

...

  middleware = applyMiddleware(
    promiseMiddleware,
    thunk,
    reduxRouterMiddleware,
    analyticsMiddleware
  );
```

The issue for me is that, I want  to send every `@@router/UPDATE_LOCATION` event to google analytics but the first page is missing.

**Comment by oliviertassinari** [comment]
For now, I'm using this fix inside at the end of my reducer list.
```js
  analyticsMiddleware({
    getState: () => state,
  })(() => {})(action);

```

**Comment by jlongster** [comment]
The middleware chain you posted does not include the `analyticsMiddleware` you have above. Can you paste the full code you're working with?

**Comment by oliviertassinari** [comment]
Sure, here is my `analyticsMiddleware.js`

```js
import {UPDATE_LOCATION} from 'redux-simple-router';
import pluginAnalytics from 'plugin/analytics';

function analyticsMiddleware(store) {
  return next => action => {
    const result = next(action);

    if (action && action.type === UPDATE_LOCATION) {
      pluginAnalytics.trackView(store.getState().get('routing').location.pathname);
    }

    return result;
  };
}

export default analyticsMiddleware;
```

**Comment by timdorr** [comment]
Ah, it's because the listener is being set up before the rest of the middleware is applied to the store and dispatching before `applyMiddleware` completes. This may actually be a redux bug....

We could hack around this with `setTimeout`, but I'm not a huge fan of that approach.

**Comment by timdorr** [comment]
Opened up rackt/redux#1240. We'll see if anything happens there.

**Comment by damassi** [comment]
This is happening to me as well after upgrading. I have an authorization / restricted routes flow and the way I'm getting around it is by triggering a fake system action on init, which will then process my middleware as it did before.

**Comment by timdorr** [comment]
Well, I guess we'll have to defer our initial listener with setTimeout. That's the only way around it I can see. I'm open to PRs if someone has a better solution. I'll try to get this tested, fixed, and released soon.

**Comment by timdorr** [comment]
Thought I had a fix, but didn't work. setTimeout just makes listening async, which works fine in the browser, but busts everything in the server context.

**Comment by timdorr** [comment]
This should be solved once #259 settles, as it operates on an instantiated store which will have all of its middleware set up.

**Comment by gabrielhpugliese** [comment]
For those who are using it, the upgrade fixed the problem? My version is 2.0.4 and I see this happening.

**Comment by oliviertassinari** [comment]
Yes it does.
